docs/developer/checkout/performance.mdx

```diff
@@ -28,4 +28,64 @@ You can leverage strategy called "optimistic updates" for that.
 - [React Query](https://tanstack.com/query/v4/docs/framework/react/guides/optimistic-updates)
 - [urql](https://commerce.nearform.com/open-source/urql/docs/graphcache/cache-updates/#optimistic-updates)
 
-In the nutshell, frontend can calculate new price locally and update the UI immediately, while the real response will apply with a delay in the background.
\ No newline at end of file
+In the nutshell, frontend can calculate new price locally and update the UI immediately, while the real response will apply with a delay in the background.
+
+### Lazy vs eager recalculations
+
+Some mutations require price recalculations - such as changing lines quantity or adding a voucher. However, not every mutation behaves the same way.
+
+With performance considerations, there are two strategies how mutations recalculate prices:
+
+- Eager: Mutation will call `*_CALCULATE_TAXES` event during the mutation execution. That means mutation will run slower and the returned value will have latest updates prices values.
+- Lazy: Mutation will mark internal prices to be "dirty", but not call the webhook immediately. Webhook will be called once the value is requested afterward.
+
+```mermaid
+sequenceDiagram
+sequenceDiagram
+    actor Storefront
+
+    Storefront->>Saleor: mutation checkoutLineUpdate { errors }
+    note left of Saleor: Note that taxable fields are not requested
+    Saleor-->>Saleor DB: Mark prices to be "dirty"
+    Saleor->>Storefront: Ok<br/> { errors: [] }
+
+    Storefront->>Saleor: mutation checkoutLineUpdate { errors }
+    Saleor-->>Saleor: Prices are dirty already
+    Saleor->>Storefront: Ok<br/> { errors: [] }
+
+    Storefront->>+Saleor: query Checkout {total { gross }}
+    note left of Saleor: Note that total.gross requires recalculation
+    create participant Taxes App
+    Saleor->>Taxes App: CHECKOUT_CALCULATE_TAXES
+    Taxes App->>Saleor: Taxes returned
+    Saleor-->>Saleor: Calculate final prices
+    Saleor->>Saleor DB: Mark prices to be "not dirty"
+
+    Saleor->>-Storefront: Ok <br/>{ total: {gross} }
+```
+
+#### Eager mutations
+
+Following Order mutations are eager:
+- draftOrderComplete
+- orderDiscountAdd
+- orderDiscountUpdate
+- orderLineDiscountUpdate
+- orderLineDiscountRemove
+
+Following Checkout mutations are eager:
+- TODO
+
+Other mutations are lazy
+
+#### Fields that cause taxes recalculations
+
+For lazy operations, only requesting certain fields will trigger prices recalculation (including synchronous webhook calls).
+
+These fields are (for Checkout & Order in Draft or Unconfirmed status)
+- `total`, `subtotal` & `undiscountedTotal`
+- `undiscountedShippingPrice`, `shippingPrice` & `shippingTaxRate`
+- lines `total` & `subtotal`
```

Comment:
Also `undiscountedTotalPrice`. 
And no `subtotal` as line does not have `subtotal` field